### PR TITLE
fixed the bugs related to relocating db files for restore

### PR DESCRIPTION
**Pull request by llali**

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseHelper.cs [hunk 1]
```diff
@@ -234,9 +240,16 @@ private void UpdateRestorePlan(RestoreDatabaseTaskDataObject restoreDataObject)
...
             }
             restoreDataObject.RestorePlanner.ReadHeaderFromMedia = !string.IsNullOrEmpty(restoreDataObject.RestoreParams.BackupFilePath);
-            var dbNames = restoreDataObject.GetSourceDbNames();
-            string dbName = dbNames.First();
-            restoreDataObject.RestorePlanner.DatabaseName = dbName;
+           
+            restoreDataObject.RestorePlanner.DatabaseName = restoreDataObject.DefaultDbName;
+            restoreDataObject.TargetDatabase = restoreDataObject.RestoreParams.DatabaseName;
+            //TODO: used for other types of restore
+            /*bool isTailLogBackupPossible = restoreDataObject.RestorePlanner.IsTailLogBackupPossible(restoreDataObject.RestorePlanner.DatabaseName);
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseHelper.cs]: dead code...I assume you plan to enable in a later iteration?

File: src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs [hunk 2]
```diff
@@ -130,7 +142,14 @@ public RestoreOptions RestoreOptions
         /// </summary>
         public string DataFilesFolder
         {
-            get { return this.dataFilesFolder; }
+            get
+            {
+                if (string.IsNullOrEmpty(this.dataFilesFolder))
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs]: this getter uses a slightly different pattern then the one you used in the other getter you added on line 44.  I'd recommend using the same pattern in general for consistency.

File: src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs [hunk 3]
```diff
@@ -247,6 +274,37 @@ private void RelocateDbFiles()
...
+        /// </summary>
+        internal void UpdateDBFilesPhysicalRelocate()
+        {
+            foreach (DbFile item in DbFiles)
+            {
+                string fileName = this.GetTargetDbFilePhysicalName(item.PhysicalName);
+                item.PhysicalNameRelocate = PathWrapper.Combine(PathWrapper.GetDirectoryName(item.PhysicalName),
+                    fileName);
+                Uri pathUri;
+                bool fUriCreated = Uri.TryCreate(item.PhysicalNameRelocate, UriKind.Absolute, out pathUri);
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs]: we shouldn't use Hungarian notation for variables...please remove the 'f' prefix.

File: src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs [hunk 4]
```diff
@@ -286,9 +344,17 @@ internal set
...
-        /// The database used to restore from
+        /// The database from the backup file used to restore to by default
         /// </summary>
-        public string sourceDbName = string.Empty;
+        public string DefaultDbName
+        {
+            get
+            {
+                var dbNames = GetSourceDbNames();
+                string dbName = dbNames.First();
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs]: dbNames will never be null or an empty list?  If that's not a true statement we should due null checking and perhaps use FirstOrDefault

File: src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseHelper.cs (hunk 1 above)
Comment (llali) [inline src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseHelper.cs]: yes it's not used for full backup

---
In reply to: [127762843](https://github.com/Microsoft/sqltoolsservice/pull/414#discussion_r127762843) [](ancestors = 127762843)

**Pull request merged**